include/ddc/kernels/splines/spline_builder.hpp

```diff
@@ -324,17 +324,29 @@ class SplineBuilder
     /**
      * @brief Get the whole domain on which spline coefficients are defined, with the dimension of interest being the leading dimension.
      *
-     * This is used internally due to solver limitation and because it may be beneficial to computation performance.
+     * This is used internally due to solver limitation and because it may be beneficial to computation performance. For LAPACK backend and non-periodic boundary condition, we are using SplinesLinearSolver3x3Blocks which requires upper_block_size additional rows for internal operations.
      *
      * @return The (transposed) domain for the spline coefficients.
      */
     batched_spline_tr_domain_type batched_spline_tr_domain() const noexcept
     {
-        return batched_spline_tr_domain_type(
-                batched_spline_domain().restrict(ddc::DiscreteDomain<bsplines_type>(
+        int upper_block_size = 0;
+        if constexpr (Solver == ddc::SplineSolver::LAPACK && !bsplines_type::is_periodic()) {
+            int lower_block_size;
+            if constexpr (bsplines_type::is_uniform()) {
+                compute_block_sizes_uniform(lower_block_size, upper_block_size);
+            } else {
+                compute_block_sizes_non_uniform(lower_block_size, upper_block_size);
+            }
+        }
+
+        return batched_spline_tr_domain_type(ddc::replace_dim_of<bsplines_type, bsplines_type>(
+                batched_spline_domain(),
+                ddc::DiscreteDomain<bsplines_type>(
                         ddc::DiscreteElement<bsplines_type>(0),
                         ddc::DiscreteVector<bsplines_type>(
-                                ddc::discrete_space<bsplines_type>().nbasis()))));
+                                ddc::discrete_space<bsplines_type>().nbasis()
+                                + upper_block_size))));
```

Comment:
I am skeptical about this change. It seems difficult to maintain because we need to keep in mind the extra rows are specific to one particular solver. If we add a new LAPACK solver we may forget to update this.

Could we imagine adding a new virtual functions to solvers that would return the expected size for `B` ?

Reply:
Similar to [`required_span_size`](https://en.cppreference.com/w/cpp/container/mdspan/layout_right/mapping) on standard layouts

Reply:
Ok I will look at it

Reply:
thx for the proposal it's far better like that